blob/src/main/java/io/crate/common/Octal.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.common;
+
+import java.util.Arrays;
+import java.util.Locale;
+
+/**
+ * Encodes and decodes binary strings into the "bytea" escape format.
+ * Unprintable bytes, outside of the 32..126 range are represented with an octal number "\nnn", and backslashes
+ * are doubled.
+ */
+public class Octal {
```

Comment:
We should move this class and the existing `Hex` one into the `shared` subproject, this new class is not used inside `blob` at all and it contains generic logic anyway.